main/acle.md

```diff
@@ -4955,6 +5111,2370 @@ assertions, which may speed up access to aligned data (and will fault
+with a scalar index:
+
+``` c
+  svint32_t svld1_gather[_u32base]_index_s32(svbool_t pg, svuint32_t bases,
+                                             int64_t index)
+```
+
+The address of element i is:
+
+``` c
+  ((int32_t *)(uintptr_t)bases[i]) + index
+```
+
+### SVE operations involving vectors and scalars
+
+Some of the SVE instructions have immediate forms; for example:
+
+``` c
+  ADD Zd.S, Zd.S, #1
+```
+
+adds 1 to every element of `Zd.S`. ACLE extends this approach to
+all arithmetic operations and to all scalar inputs (not just immediates).
+The implementation can then use immediate forms where possible or duplicate
+the scalar into a vector otherwise.
+
+For example:
+
+``` c
+  svint32_t x;
+  ...
+  x = svadd[_n_s32]_x(pg, x, 1);
+```
+
+adds 1 to every active element of `x` while:
+
+``` c
+  svfloat64_t x;
+  double *ptr;
+  ...
+  x = svadd[_n_f64]_x(pg, x, *ptr);
+```
+
+adds `*ptr` to every active element of `x`. The first example is
+likely to use the immediate form of ADD while the latter is likely to
+use LD1RD.
+
+In a vector operation, the disambiguator `_n` indicates that the final
+operand is a scalar rather than a vector.
+
+### Immediate arguments to SVE intrinsics
+
+Some intrinsics take enumeration values as arguments. These enumerations
+must always be integer constant expressions that specify a valid
+enumeration value.
+
+A few intrinsics take general integer immediates as arguments. In the
+[intrinsics documentation](https://developer.arm.com/architectures/instruction-sets/intrinsics/#f:@navigationhierarchiessimdisa=[sve,sve2]),
+these arguments always start with the prefix `imm`. Immediate arguments
+must be integer constant expressions within a certain range (which is
+the same as the range of the underlying SVE instruction).
+
+### Faults and exceptions in SVE intrinsics
+
+Loads and stores can trigger faults in the same way as normal pointer
+dereferences. Exactly what happens then depends on the host environment
+and is out of scope for this document. However, these faults should be
+symptoms of a program going wrong rather than something that the
+programmer deliberately planted. The implementation can therefore remove
+or reorder potentially faulting operations as long as:
+
+1.  doing so would not cause a previously non-faulting program to fault;
+    and
+
+2.  faults do not move between C++ exception blocks, in cases where
+    faults are reported as exceptions.
+
+Also, many floating-point operations can raise IEEE exceptions. A
+similar set of rules apply there: the implementation can remove or
+reorder operations that might raise an IEEE exception if:
+
+1.  doing so would not cause a program to raise IEEE exceptions in cases
+    where it would not previously;
+
+2.  IEEE exceptions do not move between C++ exception blocks, in cases
+    where IEEE exceptions are reported as C++ exceptions; and
+
+3.  IEEE exceptions do not move across functions that manipulate the IEEE
+    exception state.
+
```

Comment:
```suggestion
* doing so would not cause a program to raise IEEE exceptions in cases
    where it would not previously;

*  IEEE exceptions do not move between C++ exception blocks, in cases
    where IEEE exceptions are reported as C++ exceptions; and

* IEEE exceptions do not move across functions that manipulate the IEEE
    exception state.

```